src/formpack/schema/fields.py

```diff
@@ -224,8 +224,18 @@ def get_disaggregated_stats(self, metrics, top_splitters,
         not_provided = 0
         provided = 0
         for val, counter in metrics.items():
-            not_provided += counter.pop(None, 0)
-            provided += counter.pop('__submissions__', 0)
+            none_submissions = counter.pop(None, 0)
+            not_provided += none_submissions
+            # `counter[None]` corresponds to submissions with no values for current field (`self`)
+            # If `val` is None, some submissions don't have any values for the `splitted_by_field`.
+            # We should consider all these submissions as `not_provided`
+            if val is None:
+                not_provided += sum(counter.values())
+            else:
+                # `counter['__submissions__'] contains the count of all submissions,
```

Comment:
I'm having trouble figuring out where this happens. I found these:
https://github.com/kobotoolbox/formpack/blob/09c600f2ba62e1964a5fbf1bb0fcb44f79c6d750/src/formpack/reporting/autoreport.py#L161-L162
https://github.com/kobotoolbox/formpack/blob/09c600f2ba62e1964a5fbf1bb0fcb44f79c6d750/src/formpack/reporting/autoreport.py#L84-L85
...but they both increment `counter['__submissions__']` only when the value is not `None`.

Reply:
@jnm, 

Let's say, we have a form with two questions (select multiple), not mandatory:

Favorite coffee:
- Nespresso
- Keurig
- Tim Horton

Type of coffee:
- Regular
- Espresso
- Latte


Three users submit data (first case):
First one:
	- Favorite coffee: `Nespresso`
	- Type of coffee: `Regular`

Second one:
	- Favorite coffee: `Tim Horton`
	- Type of coffee: `Latte`

Third one:
	- Favorite coffee: `Keurig`
	- Type of coffee: `No choices checked`
	
Disaggregating the stats by grouping by `Favorite coffee`  should return 3 `Counter`s 
- `None` => `{'keuring':  1}`
- `Latte` => `{'tim_horton':  1}`
- `Regular` => `{'nespresso':  1}`

Now, a fourth user submits this (second case):
	- Favorite coffee: `No choices checked`
	- Type of coffee: `latte`

It should still return the 3 sames counters, except that `Latte` Counter should have changed for
`{'None': 1, 'tim_horton':  1}`

Having that said, the `if` condition handles the first case, the `else` handles the second one.

There is already a test that covers the first case: `tests/test_autoreport.py::TestAutoReport::test_disaggregate`.
It doesn't seem to have a test for second case, though. I'll add one.